src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentExtensions.cs

```diff
@@ -161,6 +161,11 @@ public static ContentElement Apply(this ContentElement contentElement, string na
             // Replace the existing content element with the new one
             contentElement.Elements[name] = element;
 
+            if (element is ContentField)
+            {
+                contentElement.ContentItem.Elements.Clear();
```

Comment:
Do you understand what it's actually doing?
Would you mind adding a line of comment about why it has to be done?

Also can you point me to where this is called from, in your case. Just for me to understand if it's actually safe to call `Clear` at that point.

Reply:
Yeah, so in my case it's being called from [UpdateEditorAsync in ContentFieldDisplayDriver](https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentFieldDisplayDriver.cs#L178).

What's happening is that when a field is updated it updates it's own entry in the `Elements` dictionary but the part has already been added to the content item's `Elements` dict. So when you call `contentItem.Get<Part>()` the part from the dictionary is returned without the field being populated.

By clearing the content item's dict when updating a field it causes the part to be re-evaluated and the field gets populated. It would probably be better to just remove the single part from dictionary but in this context we don't have the name of the part that the field belongs to so I just opted to clear the whole thing.

Reply:
Here's the section that doesn't run when an element is already in the dictionary: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentExtensions.cs#L49-L60